sycl/unittests/helpers/test/PiMock.cpp

```diff
@@ -0,0 +1,89 @@
+//==--------- PiMock.cpp --- A test for mock helper API's ------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <gtest/gtest.h>
+#include <helpers/PiMock.hpp>
+
+#include <detail/queue_impl.hpp>
+
+class PiMockTest : public ::testing::Test {};
+
+using namespace cl::sycl;
+
+pi_result
+piProgramBuildRedefine(pi_program program, pi_uint32 num_devices,
+                       const pi_device *device_list, const char *options,
+                       void (*pfn_notify)(pi_program program, void *user_data),
+                       void *user_data) {
+  return PI_INVALID_BINARY;
+}
+
+pi_result piKernelCreateRedefine(pi_program program, const char *kernel_name,
+                                 pi_kernel *ret_kernel) {
+  return PI_INVALID_DEVICE;
+}
+
+TEST(PiMockTest, ConstructFromQueue) {
+  queue NormalQ;
+  queue MockQ;
+  unittest::PiMock Mock(MockQ);
+
+  const auto &NormalPiPlugin =
+      detail::getSyclObjImpl(NormalQ)->getPlugin().getPiPlugin();
+  const auto &MockedQueuePiPlugin =
+      detail::getSyclObjImpl(MockQ)->getPlugin().getPiPlugin();
+  const auto &PiMockPlugin =
+      detail::getSyclObjImpl(Mock.getPlatform())->getPlugin().getPiPlugin();
+  EXPECT_EQ(&MockedQueuePiPlugin, &PiMockPlugin)
+      << "The mocked object and the PiMock instance must share the same plugin";
+  ASSERT_FALSE(&NormalPiPlugin == &MockedQueuePiPlugin)
+      << "Normal and mock platforms must not share the same plugin";
+}
+
+TEST(PiMockTest, ConstructFromPlatform) {
+  platform NormalPlatform(cpu_selector{});
```

Comment:
What happens with this test if only host device present? That’s the case with post commit.

Reply:
Early exit works for me